src/dpi.rs

```diff
@@ -63,9 +63,11 @@
 //! - **macOS:** "retina displays" have a scale factor of 2.0. Otherwise, the scale factor is 1.0.
 //!   Intermediate scale factors are never used. It's possible for any display to use that 2.0 scale
 //!   factor, given the use of the command line.
-//! - **X11:** On X11, we calcuate the scale factor based on the millimeter dimensions provided by
-//!   XRandR. This is almost certainly the wrong approach, but many man-hours have been spent trying
-//!   to figure out how to do it better and X11 provides no easy answer.
+//! - **X11:** Many man-hours have been spent trying to figure out how to handle DPI in X11. Winit
+//!   currently uses a three-pronged approach.
+//!   + Calcuate the scale factor based on the millimeter monitor dimensions provided by XRandR.
+//!   + Use the value set in `Xft.dpi` in Xresources. Overrides XRandR's monitor size.
+//!   + Use the value in the `WINIT_X11_SCALE_FACTOR` environment variable. Overrides `Xft.dpi` and XRandR.
```

Comment:
This is not true, WINIT_X11_SCALE_FACTOR doesn't overrides Xft.dpi. We do have a bug about it https://github.com/rust-windowing/winit/issues/1141 .

Reply:
I meant that the situation is not like we have a bug that we should fix. The situation is more like "should we override it or not", I can send patches for both #1141 and #836 , but we're kind of not sure what is fine for winit here.

Reply:
Ah, that's a weird choice. IMO `WINIT_X11_SCALE_FACTOR` should override `Xft.dpi`, since it's more specific. I'd be fine accepting a PR to fix it, and implementing the solution you described in https://github.com/rust-windowing/winit/issues/836#issuecomment-562912905.

Reply:
Then I'll send a PR to fix both of them.